pkg/gameserverallocations/allocator.go

```diff
@@ -250,15 +250,26 @@ func (c *Allocator) loggerForGameServerAllocation(gsa *allocationv1.GameServerAl
 }
 
 // allocateFromLocalCluster allocates gameservers from the local cluster.
+// Add here  a a number of times we are entering
 func (c *Allocator) allocateFromLocalCluster(ctx context.Context, gsa *allocationv1.GameServerAllocation) (*allocationv1.GameServerAllocation, error) {
 	var gs *agonesv1.GameServer
+	latency := c.newMetrics(ctx)
+	retryCount := 0
 	err := Retry(allocationRetry, func() error {
 		var err error
 		gs, err = c.allocate(ctx, gsa)
+		retryCount++
+
 		if err != nil {
-			c.loggerForGameServerAllocation(gsa).WithError(err).Warn("failed to allocate. Retrying...")
+			c.loggerForGameServerAllocation(gsa).WithError(err).Warn("Failed to Allocated. Retrying...")
+		} else {
+			latency.setError(string("Success ") + fmt.Sprint(retryCount))
```

Comment:
What's happening here? Not sure I follow.

Reply:
So if there's no error inside this retry loop it means that we had a success allocation within the 5 steps that we have defined for `allocationRetry = wait.Backoff` object. I think the method name is misleading since it's not an Error that we are setting (I'll change that) but it was a way for me to set a label and being able to query it as such: 

![image](https://github.com/user-attachments/assets/6fba3d0f-0035-4e8d-8ba4-4b34287e7fe8)

Reply:
Yeah that's not a good idea for a variety of reasons (mostly labels are not metrics). You just remade histograms 😄 

Use a histogram, and do a linear bucket from 1 to 10 (whatever the top expected number is?), and store it in that.

https://prometheus.io/docs/concepts/metric_types/#histogram

Exactly the same as duration above, but instead you're capturing how many retries.

Reply:
Ahhh okok my bad, I see what you mean, I changed it and now this is the results i get in Prometheus: 
![image](https://github.com/user-attachments/assets/cc117b4a-fa1f-4ec9-a408-209b038a7e61)

Reply:
That looks better 👍🏻